rest/servers/spring/src/main/java/com/connexta/store/rest/spring/StoreApi.java

```diff
@@ -0,0 +1,171 @@
+@Api(value = "Store")
+public interface StoreApi {
+
+  // See https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+  String SEMANTIC_VERSION_REGEX =
+      "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$";
+
+  @ApiOperation(
+      value = "Add metadata to a Dataset.",
+      nickname = "addMetadata",
+      notes = "Clients send information about metadata to add to an existing Dataset.",
+      tags = {
+        "store, promote, addMetadata",
+      })
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 200, message = "The metadata was added to the Dataset."),
+        @ApiResponse(
+            code = 400,
+            message =
+                "The client message could not be understood by the server due to invalid format or syntax.",
+            response = ErrorMessage.class),
+        @ApiResponse(
+            code = 401,
+            message = "The client could not be authenticated.",
+            response = ErrorMessage.class),
+        @ApiResponse(
+            code = 403,
+            message = "The client is not authorized.",
+            response = ErrorMessage.class),
+        @ApiResponse(
+            code = 404,
+            message = "The requested Dataset could not be found.",
+            response = ErrorMessage.class),
+        @ApiResponse(
+            code = 501,
+            message = "The requested API version is not supported and therefore not implemented.",
+            response = ErrorMessage.class)
+      })
+  @RequestMapping(
+      value = "/dataset/{datasetId}",
+      produces = {"application/json"},
+      consumes = {"application/json"},
+      method = RequestMethod.PUT)
+  @ResponseHeader(
+      name = "Content-Version",
+      description =
+          "The API version used by the server to produce the REST message. The server will accept messages for any minor versions prior to this one.",
+      response = String.class)
+  default ResponseEntity<Void> addMetadata(
+      @ApiParam(
+              value =
+                  "The API version used by the client to produce the REST message. The client must accept responses marked with any minor versions after this one. This implies that all future minor versions of the message are backward compatible with all previous minor versions of the same message.",
+              required = true)
+          @RequestHeader(value = "Accept-Version", required = true)
+          @NotNull
+          @Pattern(regexp = SEMANTIC_VERSION_REGEX)
+          String acceptVersion,
+      @Pattern(regexp = "^[0-9a-zA-Z]+$")
+          @Size(min = 32, max = 32)
+          @ApiParam(value = "The ID of the Dataset.", required = true)
+          @PathVariable("datasetId")
+          String datasetId,
+      @ApiParam(value = "file detail") @Valid @RequestPart("file") MultipartFile file)
```

Comment:
❗️ I would expect this param to be an AddMetadataRequest

Reply:
Whoops forgot to fill it in :)